### PR TITLE
Update integration tests to use REST APIs in master branch (III)

### PR DESCRIPTION
**Pull request by ashanthamara**
### application/auth
- SecondaryStoreUserLoginTestCase

### OAuth2/consented.token
- OAuth2ServiceAuthCodeGrantJWTAccessTokenWithConsentedTokenColumnTestCase
- OAuth2ServicePasswordGrantJWTAccessTokenWithConsentedTokenColumnTestCase
- OAuth2ServiceWithConsentedTokenColumnAbstractIntegrationTest

 ### oidc
- OIDCAbstractIntegrationTest
- OIDCAuthCodeGrantSSODifferentSubjectIDTestCase
- OIDCAuthCodeGrantSSOTestCase
- OIDCAuthzCodeIdTokenValidationTestCase
- OIDCFederatedIdpInitLogoutTest
- OIDCFileBasedSkipLoginConsentTestCase
- OIDCIdentityFederationTestCase
- OIDCPasswordGrantTest
- OIDCSPWiseSkipLoginConsentTestCase
- OIDCSSOConsentTestCase
- OIDCUtilTest

### saml
- ChangeACSUrlTestCase
- IDPMetadataTestCase
- RegistryMountTestCase
- SAMLErrorResponseTestCase
- SAMLFederationDynamicQueryParametersTestCase
- SAMLFederationWithFileBasedSPAndIDPTestCase

## Added following rest clients
- `UserStoreMgtRestClient`
- `TenantMgtRestClient`

## Related Issue

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/model/UserStoreReq.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        @Override
+        public String toString() {
+
+            return "class ClaimAttributeMapping {\n" +
+                    "    claimURI: " + toIndentedString(claimURI) + "\n" +
+                    "    mappedAttribute: " + toIndentedString(mappedAttribute) + "\n" +
+                    "}";
+        }
+    }
+}
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/model/UserStoreReq.java:347]: Add new line.

Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/model/UserStoreReq.java]: Fixed with https://github.com/wso2/product-is/pull/16354/commits/8796c767b55bfb43464a54ea8f1ddb369a1a1401

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/UserStoreMgtRestClient.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    /**
+     * Check user store deployment
+     *
+     * @param domain User Store name
+     * @return boolean response of the user store deployment.
+     */
+    public boolean waitForUserStoreDeployment(String domain) throws Exception {
+
+        long waitTime = System.currentTimeMillis() + 30000; //wait for 45 seconds
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/UserStoreMgtRestClient.java:107]: Comment says you want to wait 45 seconds. But you are waiting 30 seconds only.

Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/UserStoreMgtRestClient.java]: Fixed the comment with https://github.com/wso2/product-is/pull/16354/commits/2e2720d50618a38ef60a3d96e0fe0a7dd5439f26

**Review by madurangasiriwardena — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java [hunk 3]
```diff
@@ -245,4 +273,59 @@ private void stopTomcat() throws LifecycleException {
         tomcat.destroy();
         LOG.info("Tomcat server stopped.");
     }
+
+    private void addSecondaryJDBCUserStore() throws Exception {
+
+        //creating database
```
Comment (UdeshAthukorala) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java:279]: `// Create database.`

Comment (UdeshAthukorala) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java:279]: Shall we fix the formatting here? Let's fix other places as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java [hunk 4]
```diff
@@ -245,4 +273,59 @@ private void stopTomcat() throws LifecycleException {
...
+        H2DataBaseManager dbmanager = new H2DataBaseManager("jdbc:h2:" + ServerConfigurationManager.getCarbonHome()
+                + "/repository/database/" + USER_STORE_DB_NAME, DB_USER_NAME, DB_USER_PASSWORD);
+        dbmanager.executeUpdate(new File(ServerConfigurationManager.getCarbonHome() + "/dbscripts/h2.sql"));
+        dbmanager.disconnect();
+
+        // Register a secondary user store
```
Comment (UdeshAthukorala) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java:285]: Shall we add the missing full stop here?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/consented/token/OAuth2ServiceAuthCodeGrantJWTAccessTokenWithConsentedTokenColumnTestCase.java [hunk 5]
```diff
@@ -111,9 +111,9 @@ public void testSendAuthorizedPost() throws Exception {
         urlParameters.add(new BasicNameValuePair("authorizeEndpoint", OAuth2Constant.APPROVAL_URL));
         urlParameters.add(new BasicNameValuePair("authorize", OAuth2Constant.AUTHORIZE_PARAM));
         // email scope is to retrieve the email address of the user.
-        // phone scope is to retrieve the telephone number of the user.
+        // address scope is to retrieve the country number of the user.
```
Comment (UdeshAthukorala) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/consented/token/OAuth2ServiceAuthCodeGrantJWTAccessTokenWithConsentedTokenColumnTestCase.java:114]: Let's start the comment with a capital letter. Let's fix the other places as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java (hunk 3 above)
Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java:279]: Fixed with https://github.com/wso2/product-is/pull/20096

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java (hunk 4 above)
Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java:285]: Fixed with https://github.com/wso2/product-is/pull/20096

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/consented/token/OAuth2ServiceAuthCodeGrantJWTAccessTokenWithConsentedTokenColumnTestCase.java (hunk 5 above)
Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/consented/token/OAuth2ServiceAuthCodeGrantJWTAccessTokenWithConsentedTokenColumnTestCase.java:114]: Fixed with https://github.com/wso2/product-is/pull/20096

**Pull request merged**